numpy/__init__.pyi

```diff
@@ -181,12 +225,122 @@ class ndarray:
     @property
     def ndim(self) -> int: ...
 
+    # Many of these special methods are irrelevant currently, since protocols
+    # aren't supported yet. That said, I'm adding them for completeness.
+    # https://docs.python.org/3/reference/datamodel.html
+    def __len__(self) -> int: ...
+    def __getitem__(self, key) -> Any: ...
+    def __setitem__(self, key, value): ...
+    def __iter__(self) -> Any: ...
+    def __contains__(self, key) -> bool: ...
+
+    def __int__(self) -> int: ...
+    def __float__(self) -> float: ...
+    def __complex__(self) -> complex: ...
+    if sys.version_info[0] < 3:
+        def __oct__(self) -> str: ...
+        def __hex__(self) -> str: ...
+        def __nonzero__(self) -> bool: ...
+    else:
+        def __bool__(self) -> bool: ...
+    def __bytes__(self) -> bytes: ...
```

Comment:
I imagine this is python 3 only?

Also `__unicode__`, `__str__`, and `__repr__` are missing

Reply:
done